src/Uno.Extensions.Navigation.UI/RouteNotifier.cs

```diff
@@ -17,9 +17,11 @@ public RouteNotifier(ILogger<RouteNotifier> logger)
 	private IDictionary<Guid, StringBuilder> navigationSegments = new Dictionary<Guid, StringBuilder>();
 	private IDictionary<Guid, int> runningNavigations = new Dictionary<Guid, int>();
 
+	private IRegion? _initialRegion;
 
 	public void StartNavigation(INavigator navigator, IRegion region, NavigationRequest request)
 	{
+		_initialRegion = region;
```

Comment:
We would need to correlate the region to the specific NavigationRequest, no? We could have multiple navigation requests and we need to know which root region to use when EndNavigation is called.